### DISCUSSION TITLE
Kotlin examples and Kotlin support

### DISCUSSION THREAD

**Discussion by kenichi-fukushima** [line/armeria#5229] [Q&A] [answered]
Hello Armeria developers,

There are mentions of Kotlin examples here and there in Armeria documentation but I find none in https://github.com/line/armeria-examples. Is there any other place where I can find example code for common tasks in Armeria-Kotlin?

A particular task that I have is dispatching Kotlin coroutines from non-suspending Armeria methods such as `AbstractHttpService.doGet()` and `SimpleDecoratingHttpService.serve()`? I investigated how annotated services are couroutine-aware and found `ArmeriaCoroutineUtil.callKotlinSuspendingMethod()`. I might do the same thing in my code but a Kotlin beginner can't be confident that that's the right approach.

**Comment by jrhee17** [comment]
Sorry about the late response 😅 

> There are mentions of Kotlin examples here and there in Armeria documentation but I find none in https://github.com/line/armeria-examples. Is there any other place where I can find example code for common tasks in Armeria-Kotlin?

Hi, there has been a small issue with kotlin examples being exported to the `armeria-examples` repository. For now, kotlin related examples live in the main source. 
I'll try to get the dead links fixed 😄   Thanks for the notice.

https://github.com/line/armeria/tree/main/examples

> I investigated how annotated services are couroutine-aware and found ArmeriaCoroutineUtil.callKotlinSuspendingMethod(). I might do the same thing in my code but a Kotlin beginner can't be confident that that's the right approach.

Armeria internally already calls the method if the annotated service method is a suspending function, so it should work out of the box.

https://github.com/line/armeria/blob/d9b6142ecc673e1690a963d7ce8920e6272d98da/examples/annotated-http-service-kotlin/src/main/kotlin/example/armeria/server/annotated/kotlin/ContextAwareService.kt#L16-L30

**Reply by fukushima-kenichi-mf** [reply]
Thanks for the reply. I will check those example in the main repo.

> Armeria internally already calls the method if the annotated service method is a suspending function, so it should work out of the box.

I actually wanted know how I could generalize what `AnnotatedService` does to other Armeria APIs that are not coroutine aware.

**Reply by jrhee17** [reply]
> I actually wanted know how I could generalize what AnnotatedService does to other Armeria APIs that are not coroutine aware.

I see, do you mind if I ask what armeria APIs you were looking to make coroutine aware?

**Reply by fukushima-kenichi-mf** [reply]
> I see, do you mind if I ask what armeria APIs you were looking to make coroutine aware?

Generic `AbstractHttpService` and decorators.

**Reply by minwoox** [reply, marked as answer]
You might want to refer to this example:
https://github.com/line/armeria/issues/4725#issuecomment-1458439130
We've also added an extension method for context propagation which will be contained in the next release(1.26.0).

**Reply by fukushima-kenichi-mf** [reply]
Thanks! That seems like what exactly I was looking for.
